components/abstractions/src/main/java/com/microsoft/kiota/serialization/SerializationWriterFactoryRegistry.java

```diff
@@ -36,20 +38,92 @@ public SerializationWriterFactoryRegistry() {
         if (contentType.isEmpty()) {
             throw new NullPointerException("contentType cannot be empty");
         }
-        final String vendorSpecificContentType = contentType.split(";")[0];
+        final ContentTypeWrapper contentTypeWrapper = new ContentTypeWrapper(contentType);
```

Comment:
While I appreciate the effort to refactor things, this is going to introduce new allocations on a method that's on the hot path.

Couldn't this method simply call into the new method with a default value to avoid duplication?